onnxruntime/core/framework/tensorprotoutils.cc

```diff
@@ -1034,41 +1113,42 @@ template common::Status GetSizeInBytesFromTensorProto<kAllocAlignment>(const ONN
-        element_count = initializer.DATA_SIZE();                                           \
-        tensor_byte_size = element_count * sizeof(ELEMENT_TYPE);                           \
-      }                                                                                    \
-      tensor_data_length = tensor_byte_size;                                               \
-      unpacked_tensor.reset(new uint8_t[tensor_data_length]);                              \
-      return onnxruntime::utils::UnpackTensor(                                             \
-          initializer,                                                                     \
-          initializer.has_raw_data() ? initializer.raw_data().data() : nullptr,            \
-          initializer.has_raw_data() ? initializer.raw_data().size() : 0,                  \
-          reinterpret_cast<ELEMENT_TYPE*>(unpacked_tensor.get()), element_count);          \
-    }                                                                                      \
-    break;                                                                                 \
+#define CASE_UNPACK(TYPE, ELEMENT_TYPE, DATA_SIZE)                              \
+  case ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_##TYPE: {     \
+    size_t element_count = 0;                                                   \
+    if (initializer.has_raw_data()) {                                           \
+      tensor_byte_size = initializer.raw_data().size();                         \
+      element_count = tensor_byte_size / sizeof(ELEMENT_TYPE);                  \
+    } else {                                                                    \
+      element_count = initializer.DATA_SIZE();                                  \
+      tensor_byte_size = element_count * sizeof(ELEMENT_TYPE);                  \
+    }                                                                           \
+    tensor_data_length = tensor_byte_size;                                      \
+    unpacked_tensor.reset(new uint8_t[tensor_data_length]);                     \
+    return onnxruntime::utils::UnpackTensor(                                    \
+        initializer,                                                            \
+        initializer.has_raw_data() ? initializer.raw_data().data() : nullptr,   \
+        initializer.has_raw_data() ? initializer.raw_data().size() : 0,         \
+        reinterpret_cast<ELEMENT_TYPE*>(unpacked_tensor.get()), element_count); \
+    break;                                                                      \
   }
 
 Status UnpackInitializerData(const onnx::TensorProto& initializer,
                              const Path& model_path,
                              std::unique_ptr<uint8_t[]>& unpacked_tensor,
                              size_t& tensor_data_length) {
   SafeInt<size_t> tensor_byte_size = tensor_data_length;
```

Comment:
IMO, the name "tensor_data_length" is not obviously a length in bytes. maybe tensor_raw_data_length? perhaps this is beyond the scope of this PR.

Reply:
I think that's just to enable wrapping it in a SafeInt in the .cc so that all the real uses involve 'tensor_byte_size'. The .h file calls it tensor_byte_size as do all the calling sites. Could change the param in the .cc to something like tensor_byte_size_in.

Reply:
Renamed a few places to clarify things.